scripts/publish_windows_ssm_parameter.sh

```diff
@@ -0,0 +1,201 @@
+#!/bin/bash
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# 	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+set -uo pipefail
+
+#############################################################################################################
+# This script would be executed from a codebuild project which would set the following environment variables-
+# ACTION -> Action to perform. Can be amongst "publish-ssm-parameters", or "sync-ssm-parameter"
+# SSM_PARAM_NAMESPACE -> The SSM Parameter namespace to be used.
+# ACCOUNT_ID -> Account ID where the images are present.
+# REGIONS_TO_PUBLISH -> A comma separated list of regions where we need to publish SSM parameter.
+# PUBLIC_REGISTRY_ALIAS -> Used when syncing the SSM params.
+# PUBLIC_REPOSITORY -> Public repository which is used for the image sync.
+#############################################################################################################
+
+#############################################################################################################
+# Helper methods to publish the SSM Parameter.
+#############################################################################################################
+
+# This method publishes the SSM parameter for a given tag if the same is not present or obsolete.
+publish_ssm_param_if_obsolete() {
```

Comment:
nit: I feel like obsolete is not the best word here... missing, not present, or backfill are words that to me more clearly fit here

Reply:
Addressed this comment in the latest revision